applications/leaderboard/step6/Program.cs

```diff
@@ -169,13 +169,12 @@ enum ExitCode : int
                 $"Data Source=projects/{projectId}/instances/{instanceId}"
                 + $"/databases/{databaseId}";
 
-            using (TransactionScope scope = new TransactionScope(
-               TransactionScopeAsyncFlowOption.Enabled))
+            Int64 numberOfPlayers = 0;
+            using (var connection = new SpannerConnection(connectionString))
             {
-                Int64 numberOfPlayers = 0;
-                using (var connection = new SpannerConnection(connectionString))
+                await connection.OpenAsync();
+                await connection.RunWithRetriableTransactionAsync(async (transaction) =>
                 {
-                    await connection.OpenAsync();
                     // Execute a SQL statement to get current number of records
                     // in the Players table to use as an incrementing value 
                     // for each PlayerName to be inserted.

```

Comment:
And same comments as in previous files.

Why are these all the same?

Reply:
This sample includes extra directories step4, step5, and step6 that contain partial versions of this sample app. These directories are intended to provide guidance as part of a separate Codelab walk-through where the application is built in the following stages that correspond to the steps in Codelab:

https://codelabs.developers.google.com/codelabs/cloud-spanner-leaderboard?hl=en#0